near-sdk/src/store/vec/iter.rs

```diff
@@ -156,3 +156,113 @@ where
         Some(self.get_mut(idx).unwrap_or_else(|| env::panic_str(ERR_INDEX_OUT_OF_BOUNDS)))
     }
 }
+
+/// A draining iterator for [`Vector<T>`].
+#[derive(Debug)]
+pub struct Drain<'a, T>
+where
+    T: BorshSerialize + BorshDeserialize,
```

Comment:
just to note, but what's your opinion on getting rid of these trait bound declarations on the struct itself? I see that `Vector` and subsequently other types have trait bounds underneath for `Borsh*`, but it seems like they're really a hard requirement until the `impl` block. I just feel like it'd be cleaner if the struct itself didn't have these, but also would be fine if they stayed

Reply:
I believe it was because of the drop impl on the vec reference, but I can't think of why deserialize would be needed, I'll try to remove it now to check